path_rotate_test.go

```diff
@@ -198,3 +204,88 @@ func TestManualRotate(t *testing.T) {
 		}
 	})
 }
+
+type failingRollbackClient struct {
+	count    int
+	maxCount int
+	password string
+}
+
+func (f *failingRollbackClient) UpdateDNPassword(conf *client.Config, dn string, newPassword string) error {
+	f.count += 1
+	if f.count >= f.maxCount {
+		f.password = newPassword
+		return nil
+	}
+	return fmt.Errorf("some error")
+}
+
+func (f *failingRollbackClient) UpdateUserPassword(conf *client.Config, user, newPassword string) error {
+	panic("nope")
+}
+
+func (f *failingRollbackClient) Execute(conf *client.Config, entries []*ldif.Entry, continueOnError bool) error {
+	panic("nope")
+}
+
+var _ ldapClient = (*failingRollbackClient)(nil)
+
+func TestRollback(t *testing.T) {
```

Comment:
nit: consider renaming to `Test_rollbackPassword` or `TestRollbackPassword` to match the name of the function being tested here.